blocks/logic.ts

```diff
@@ -291,11 +283,29 @@ Extensions.register(
     'logic_op_tooltip',
     Extensions.buildTooltipForDropdown('OP', TOOLTIPS_BY_OP));
 
+/** Type of a block that has CONTROLS_IF_MUTATOR_MIXIN */
+type IfBlock = Block&IfMixin;
+interface IfMixin extends IfMixinType {}
+type IfMixinType = typeof CONTROLS_IF_MUTATOR_MIXIN;
+
+// Types for quarks defined in JSON.
+/** Type of a controls_if_if (if mutator container) block. */
+interface ContainerBlock extends Block {}
+
+/** Type of a controls_if_elseif or controls_if_else block. */
+interface ClauseBlock extends Block {
+  valueConnection_: Connection|null;
+  statementConnection_: Connection|null;
+}
```

Comment:
Should this technically be `Connection|null|undefined`?

Reply:
Hmm.  Looks like it, though I've annotated them as optional properties, which are equivalent (`valueConnection_?: Connection|null`) because that seems slightly clearer.  Unfortunately this causes a considerable number of knock-on type changes.  :-/

Reply:
(OK, not _that_ considerable.)